File: src/main/java/com/example/demo/controller/MessageController.java
package com.example.demo.controller;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.demo.service.GeminiService;

import reactor.core.publisher.Flux;

@RestController
@RequestMapping("/api/message")
public class MessageController {
    private final GeminiService geminiService;

    public MessageController(GeminiService geminiService) {
        this.geminiService = geminiService;
    }

    @PostMapping(consumes = MediaType.TEXT_PLAIN_VALUE, produces = MediaType.TEXT_EVENT_STREAM_VALUE)
    public Flux<String> message(@RequestBody String message) {
        return geminiService.streamMessage(message, null);
    }
} 

File: src/main/kotlin/com/example/demo/MessageController.kt
package com.example.demo

import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Flux

@RestController
class MessageController(
    private val geminiService: GeminiService
) {
    @PostMapping("/message", consumes = [MediaType.TEXT_PLAIN_VALUE], produces = [MediaType.TEXT_EVENT_STREAM_VALUE])
    fun message(@RequestBody message: String): Flux<String> {
        return geminiService.streamMessage(message)
    }
} 